packages/cozy-stack-client/src/DocumentCollection.js

```diff
@@ -260,6 +262,15 @@ class DocumentCollection {
     let resp
     try {
       resp = await this.fetchDocumentsWithMango(path, selector, options)
+      if (
+        resp.warning &&
+        options.partialFilter &&
+        isIndexNotUsedWarning(resp.warning)
+      ) {
+        // This warning might happen when an index including a partial filter
+        // is not created yet.
+        throw new Error('no_index')
```

Comment:
maybe we should use a more precise error message:
```suggestion
        throw new Error('no index including a partial filter created')
```

Reply:
No, this is the error produced by the stack when there is [no usable index](https://github.com/cozy/cozy-stack/blob/ac45c1e8f685cceb9fd0aa9dd44937f95368bb69/pkg/couchdb/errors.go#L231), here we generate the same error to be consistent and trap it thanks to this helper:  https://github.com/cozy/cozy-client/blob/6ef72b9296f796afac002d25cd73079a85fa9fa2/packages/cozy-stack-client/src/DocumentCollection.js#L275

Reply:
We can't use your suggestion. The goal is to run this method https://github.com/cozy/cozy-client/blob/6ef72b9296f796afac002d25cd73079a85fa9fa2/packages/cozy-stack-client/src/DocumentCollection.js#L278-L279 

So we need to throw something passing this test: https://github.com/cozy/cozy-client/blob/6ef72b9296f796afac002d25cd73079a85fa9fa2/packages/cozy-stack-client/src/Collection.js#L29-L31 . 

This `no_index` is raised by couch / stack

Reply:
I win by speed, @Crash-- wins by completeness ;)